app/src/components/assets/Asset.tsx

```diff
@@ -0,0 +1,165 @@
+import { faBan, faFlag } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IonCard, IonCardContent, IonCardHeader, IonCardSubtitle, IonCardTitle, IonCol, IonContent, IonItem, IonLabel, IonList, IonRow, useIonViewWillLeave } from "@ionic/react";
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router";
+import Api from "../../controllers/Api";
+import { s3url, fileExtensionToIcon, formatSize } from "../../globals/functions";
+import { AssetData, AssetTypeData } from "../../globals/interfaces";
+import Page from "../../pages/Page";
+
+const Asset = () => {
+    let { type, asset } = useParams<{type: string, asset: string}>();
+    const [assetType, setAssetType] = useState<AssetTypeData>({tags: [], thumnails: [], files: [], fields: []});
+    const [thisAsset, setThisAsset] = useState<AssetData>({flagsblocks: {BLOCK:[], FLAG:[], COUNT:{BLOCK: 0, FLAG: 0}}});
+    let cancelToken = axios.CancelToken.source();
+
+    //get data
+    useEffect(() => {
+        async function getAssets() {
+            //reset base values - needed so old data not shown
+            setAssetType({tags: [], thumnails: [], files: [], fields: []});
+            setThisAsset({flagsblocks: {BLOCK:[], FLAG:[], COUNT:{BLOCK: 0, FLAG: 0}}});
+            //query api
+            const fetchedAssets = await Api("assets/list.php", {"assetTypes_id": type,"all": true}, cancelToken.token);
```

Comment:
Sorry if this is a nitpick but I'd be keen to keep the lazy loading from the current app if possible. It's a lot of data to download for a device on say 3g or of a lower spec

Reply:
for sure, didn’t like this code currently but am keen to get the basics there and sort things like lazy load later. Happy to sort sooner if you think it’s a priority!

Reply:
Has been implemented